src/lnd_compatibility.rs

```diff
@@ -0,0 +1,65 @@
+mod internal {
+    #![allow(clippy::enum_variant_names)]
+    #![allow(clippy::unnecessary_lazy_evaluations)]
+    #![allow(clippy::useless_conversion)]
+    #![allow(clippy::never_loop)]
+    #![allow(clippy::uninlined_format_args)]
+
+    include!(concat!(env!("OUT_DIR"), "/configure_me_config.rs"));
+}
+
+use tonic_lnd::verrpc::VersionRequest;
+use internal::*;
+use crate::lnd::{get_lnd_client, LndCfg };
+use log::error;
+
+// On startup error out if the following checks don't pass:
+// - That GetVersion has peersrpc, signerrpc and dev build tags.
+// - That the version is > v0.16.2
+
+pub struct Compatibility;
+
+impl Compatibility {
+
+    pub async fn check_compatibility() {
+
+
+        let lnd_config = Config::including_optional_config_files(&["./lndk.conf"])
+            .unwrap_or_exit()
+            .0;
+
+        let args = LndCfg::new(lnd_config.address, lnd_config.cert, lnd_config.macaroon);
+        let mut client = get_lnd_client(args).expect("failed to connect");
+
+        let version_info = client
+            .versioner()
+            .get_version(VersionRequest {})
+            .await
+            .expect("failed to get version")
+            .into_inner();
+
+        let version = &version_info.version;
+        let tags = version_info.build_tags;
+
+        let min_version = String::from("0.16.2-beta");
+
+        if version < &min_version {
```

Comment:
Are we sure that string comparison will always follow semantic versioning? Worried that adding `-rc` (or some other tag) may mess this up?